test/unittests/execute_call_test.cpp

```diff
@@ -676,16 +802,48 @@ TEST(execute_call, call_via_imported_infinite_recursion)
         "0061736d010000000105016000017f020b01036d6f6403666f6f0000030201000a0601040010000b");
 
     const auto module = parse(wasm);
-    auto host_foo = [](Instance& instance, const Value* args, int depth) -> ExecutionResult {
+    int counter = 0;
+    auto host_foo = [&counter](Instance& instance, const Value* args, int depth) {
         // Function $f will increase depth. This means each iteration goes 2 steps deeper.
-        EXPECT_LE(depth, MaxDepth - 1);
+        EXPECT_LE(depth, MaxDepth);
+        ++counter;
+        return execute(instance, 1, args, depth + 1);
+    };
+    const auto host_foo_type = module->typesec[0];
+
+    auto instance = instantiate(*module, {{host_foo, host_foo_type}});
+
+    counter = 0;
+    EXPECT_THAT(execute(*instance, 0, {}), Traps());
+    EXPECT_EQ(counter, MaxDepth / 2 + 1);
+}
+
+TEST(execute_call, call_imported_interleaved_infinite_recursion_start_with_wasm)
```

Comment:
I think this could have been just an additional execution in the previous test?

Reply:
Merged.